### ISSUE TITLE
Is the version info binary language ID supposed to be 0?

### ISSUE THREAD

**Issue by vefatica** [microsoft/terminal#15927]
WindowsTerminal.exe's version info resource has this block header (from devenv).

![image](https://github.com/microsoft/terminal/assets/61856645/17504e15-4503-40bc-9981-d449045ca705)

And it has this binary data in the translation entry (from devenv).

![image](https://github.com/microsoft/terminal/assets/61856645/400e0065-a340-496e-a243-1be189ef0067)

In particular the binary language identifier is 0x0000.  That makes VerQueryValueW queries like this

`wsprintf(szQuery, L"\\StringFileInfo\\%04x%04x\\ProductVersion", lpTranslate[i].wLanguage, lpTranslate[i].wCodePage);`

fail when the language info comes from `VerQueryValueW(pData, L"\\VarFileInfo\\Translation", (LPVOID*) &lpTranslate, &ulen);`

because lpTranslate[i].wLanguage is 0.

My programming was based on the sample in the help for VerQueryValueW.

**Comment by lhecker** [comment]
You can query with `0x0409` (en-US) when the language is 0. I believe we aren't the only application which does this.

I'm not entirely sure how our versioning works exactly, but I think it's done by an external tool (provided by other Microsoft teams):
https://github.com/microsoft/terminal/blob/5fb2518117ddf14e05f67eed47e6e4d940a3c025/custom.props#L1-L11

**Comment by vefatica** [comment]
> You can query with 0x0409 (en-US) when the language is 0. I believe we aren't the only application which does this.

That's what I'm doing but I don't think it should be necessary.  I looked pretty hard for another app that does that ... didn't find one.

**Comment by lhecker** [comment]
Yeah I agree. I marked it up as a bug, because this is definitely pretty weird. The official documentation for these fields also doesn't make any mention of 0 being a valid value.

**Comment by DHowett** [comment]
I looked into this once! There's a bug or something in the Xes versioning tasks.

I think we just chose to work around it:

https://github.com/microsoft/terminal/blob/60843faa9e41944c0cc64c29e19657a6b21439fd/src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp#L1068-L1081

**Comment by lhecker** [comment]
> language or code page independent

I bet it means "ASCII" (ironic, I know).

> AFAICT, only "040904B0" will work in VerQueryValue()'s query string.

Yeah exactly, and I think if we put 0 for language into `\VarFileInfo\Translation` we also need to use `\StringFileInfo\000004B0\ProductVersion` for the version, or otherwise it's not consistent. The applications I linked above do this.

> P.S., what does this mean?

GitHub seems to have an outage right now. It returns a lot of 500 status codes when submitting stuff.

**Comment by vefatica** [comment]
Hmmm!  The copyright symbol (©) is in the "LegalCopyright" string.  I don't think that's ASCII.

**Comment by lhecker** [comment]
Yep, you see... That's because I'm dumb. 😄
If I had checked our own implementation I would've seen that it's UTF-16:
https://github.com/microsoft/terminal/blob/c5e0908b98573c9fccccb0dd8e4189c5a0769a5e/src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp#L1077
